go/libraries/doltcore/env/multi_repo_env.go

```diff
@@ -0,0 +1,193 @@
+	dbName := "dolt"
+	filePrefix := dbfactory.FileScheme + "://"
+
+	if strings.HasPrefix(dEnv.urlStr, filePrefix) {
+		path, err := dEnv.FS.Abs(dEnv.urlStr[len(filePrefix):])
+
+		if err == nil {
+			if strings.HasSuffix(path, dbfactory.DoltDataDir) {
+				path = path[:len(path)-len(dbfactory.DoltDataDir)]
+			}
+
+			if path[len(path)-1] == '/' {
+				path = path[:len(path)-1]
+			}
+
+			idx := strings.LastIndex(path, "/")
+			dirName := path[idx+1:]
+			dbName = dirToDBName(dirName)
+		}
+	}
+
+	mrEnv := make(MultiRepoEnv)
+	mrEnv.AddEnv(dbName, dEnv)
+
+	return mrEnv
+}
+
+// LoadMultiEnv takes a variable list of EnvNameAndPath objects loads each of the environments, and returns a new
+// MultiRepoEnv
+func LoadMultiEnv(ctx context.Context, hdp HomeDirProvider, fs filesys.Filesys, version string, envNamesAndPaths ...EnvNameAndPath) (MultiRepoEnv, error) {
+	nameToPath := make(map[string]string)
+	for _, nameAndPath := range envNamesAndPaths {
+		existingPath, ok := nameToPath[nameAndPath.Name]
+
+		if ok {
+			if existingPath == nameAndPath.Path {
+				continue
+			}
+
+			return nil, fmt.Errorf("databases at paths '%s' and '%s' both attempted to load with the name '%s'", existingPath, nameAndPath.Path, nameAndPath.Name)
+		}
+
+		nameToPath[nameAndPath.Name] = nameAndPath.Path
+	}
+
+	mrEnv := make(MultiRepoEnv)
+	for name, path := range nameToPath {
+		absPath, err := fs.Abs(path)
+
+		if err != nil {
+			return nil, err
+		}
+
+		dEnv := Load(ctx, hdp, fs, "file://"+absPath, version)
+
+		if dEnv.RSLoadErr != nil {
+			return nil, fmt.Errorf("error loading environment '%s' at path '%s': %s", name, absPath, dEnv.RSLoadErr.Error())
+		} else if dEnv.DBLoadError != nil {
+			return nil, fmt.Errorf("error loading environment '%s' at path '%s': %s", name, absPath, dEnv.DBLoadError.Error())
+		} else if dEnv.CfgLoadErr != nil {
+			return nil, fmt.Errorf("error loading environment '%s' at path '%s': %s", name, absPath, dEnv.CfgLoadErr.Error())
+		}
+
+		mrEnv.AddEnv(name, dEnv)
+	}
+
+	return mrEnv, nil
+}
+
+// LoadMultiEnvFromDir looks at each subfolder of the given path as a Dolt repository and attempts to return a MultiRepoEnv
+// with initialized environments for each of those subfolder data repositories.
+func LoadMultiEnvFromDir(ctx context.Context, hdp HomeDirProvider, fs filesys.Filesys, path, version string) (MultiRepoEnv, error) {
+	var envNamesAndPaths []EnvNameAndPath
+	err := fs.Iter(path, false, func(path string, size int64, isDir bool) (stop bool) {
+		if isDir {
+			dirName := filepath.Base(path)
+			name := dirToDBName(dirName)
+			envNamesAndPaths = append(envNamesAndPaths, EnvNameAndPath{Name: name, Path: path})
+		}
+
+		return false
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return LoadMultiEnv(ctx, hdp, fs, version, envNamesAndPaths...)
+}
+
+func dirToDBName(dirName string) string {
+	dbName := strings.TrimSpace(dirName)
+	dbName = strings.Map(func(r rune) rune {
+		if unicode.IsSpace(r) || r == '-' {
```

Comment:
Probably want to be more aggressive than this as it will break saved queries if we change it later... is there a way you can ask if it's not a letter or numeral that works for non-latin char sets?

Reply:
Disabled saved queries for multi repo for now.  Think there are a few questions there including this one.